update_group.go

```diff
@@ -0,0 +1,35 @@
+package goflags
+
+import "fmt"
+
+type (
+	UpdateCheckCallback func()
+	UpdateCallback      func()
+
+	UpdateGroup struct {
+		Options options
+		Actions actions
+	}
+
+	options struct {
+		IsUpdate, DisableUpdate bool
+	}
+
+	actions struct {
+		UpdateCallback
+		UpdateCheckCallback
+	}
+)
+
+// NewUpdateGroup creates self-update options
+func (flagset *FlagSet) NewUpdateGroup(toolName string, uc UpdateCheckCallback, u UpdateCallback) {
+	ug := UpdateGroup{
+		Options: options{IsUpdate: false, DisableUpdate: false},
+		Actions: actions{UpdateCallback: u, UpdateCheckCallback: uc},
+	}
+	flagset.CreateGroup("update", "Update",
+		flagset.BoolVarP(&ug.Options.IsUpdate, "update", "up", false, fmt.Sprintf("update %v to the latest released version", toolName)),
+		flagset.BoolVarP(&ug.Options.DisableUpdate, "disable-update-check", "duc", false, "disable automatic update check"),
+	)
```

Comment:
@RamanaReddy0M , suggesting multiple changes

- [ ] rename `NewUpdateGroup` -> `CreateUpdateGroup` which is more consistent
- [ ] `CreateUpdateGroup` should be uniform with other creategroup methods i.e should accept extra flags

```go
CrateUpdateGroup(toolName string, uc UpdateCheckCallback,u UpdateCallback,flags ...*goflags.FlagData)
```
- [ ] Rename Method parameters/options. its Final form should look like
```go
CreateUpdateGroup(toolName string,checkUpdate checkUpdateCallback, update UpdateCallback,flags ...*goflags.FlagData)
```